internal-services/catalog/file-updates-process-file-updates-task.yaml

```diff
@@ -0,0 +1,117 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: task-process-file-updates
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: release
+spec:
+  description: >-
+      Update files in a Git repository
+  params:
+      - name: fileUpdates
+        type: string
+        description: fileUpdates json string
+  results:
+      - name: fileUpdatesInfo
+        description: fileUpdates detailed information
+      - name: fileUpdatesState
+        description: fileUpdates state
+  workspaces:
+      - name: temp
+        description: Temporary working directory
+  steps:
+      - name: perform-updates
+        image: quay.io/hacbs-release/release-utils@sha256:b850b1a4e93d16050eefdb967b7875c17d7af4e86ecb0b268674dc6c3e8d90b1
+        env:
+          - name: GITLAB_HOST
+            valueFrom:
+                secretKeyRef:
+                    name: file-updates-secret
+                    key: gitlab_host
+          - name: ACCESS_TOKEN
+            valueFrom:
+                secretKeyRef:
+                    name: file-updates-secret
+                    key: gitlab_access_token
+          - name: GIT_AUTHOR_NAME
+            valueFrom:
+                secretKeyRef:
+                    name: file-updates-secret
+                    key: git_author_name
+          - name: GIT_AUTHOR_EMAIL
+            valueFrom:
+                secretKeyRef:
+                    name: file-updates-secret
+                    key: git_author_email
+        script: |
+            #!/usr/bin/env sh
+            set -eo pipefail
+
+            # loading git and gitlab functions
+            . /home/utils/gitlab-functions
+            . /home/utils/git-functions
+
+            gitlab_init
+
+            # saves the fileUpdates raw json to a file
+            TEMP=$(mktemp).json
+            cat > $TEMP <<JSON
+            $(params.fileUpdates)
+            JSON
+
+            REPOSITORY=$(jq -cr '[.repo, .ref] | join(" ")' $TEMP)
+            read REPO REVISION <<< $REPOSITORY
+
+            FILES=$(jq -cr '.paths[] | .path' $TEMP)
+
+            echo -e "=== UPDATING ${REPO} ON BRANCH ${REVISION} ===\n"
+            cd $(workspaces.temp.path)
+            git_clone_and_checkout --repository $REPO --revision $REVISION
+
+            for file in ${FILES}; do
```

Comment:
It's not great that we dig up the paths from inside the `paths` array and then in the for cycle we are looking up the array items again. It could also cause issues if a single path has several items (which shouldn't be the case, but it could happen).

Since this is an array, I would suggest using indices to access the items in the for cycle.
```sh
PATHS_LENGTH=$(jq '.paths | length' $TEMP)
for PATH_INDEX in seq 0 $((PATHS_LENGTH-1)); do
```

Reply:
changed.

Reply:
Much better, thanks.